saleor/graphql/shop/resolvers.py

```diff
@@ -0,0 +1,35 @@
+from ...core.taxes import display_gross_prices
+from ...plugins.manager import get_plugins_manager
+from ...shipping.models import ShippingMethod
+from ..channel import ChannelContext
+
+
+def resolve_available_shipping_methods(channel_slug: str, address):
+    available = ShippingMethod.objects.filter(
+        channel_listings__channel__slug=channel_slug
+    )
+    if address and address.country:
+        available = available.filter(
+            shipping_zone__countries__contains=address.country,
+        )
+
+    if available is None:
+        return []
+    manager = get_plugins_manager()
+    display_gross = display_gross_prices()
+    for shipping_method in available:
+        shipping_channel_listing = shipping_method.channel_listings.get(
```

Comment:
N+1 issue. It would be better to create some dict with this data so that we don't query channel listings in a loop.

Reply:
I refactored the function.

Reply:
`manager.apply_taxes_to_shipping(shipping_price, address)` will produce a lot of queries (until we finish the refactor started by @maarcingebala) and hits to external providers like Avalara.